torch/cuda/amp/autocast_mode.py

```diff
@@ -140,7 +140,8 @@ def decorate_autocast(*args, **kwargs):
 # may be falsely detected as "Iterables."
 def _cast(value, dtype):
     if isinstance(value, torch.Tensor):
-        return value.to(dtype) if (value.is_floating_point() and value.is_cuda) else value
+        is_eligible = (value.is_floating_point() and value.is_cuda and (value.dtype is not torch.float64))
```

Comment:
why not float64? No implicit downcasts?

Reply:
I'd like the behavior of custom functions to align with natively cast-listed functions, so I'm using the [same eligibility criteria](https://github.com/pytorch/pytorch/blob/ce3555a6351feb0550846ab8e99b4e0505fe3436/aten/src/ATen/autocast_mode.cpp#L142) for input casting.